values.go

```diff
@@ -55,10 +56,15 @@ func NewVoid() Void {
 	return Void{}
 }
 
+func NewMeteredVoid(memoryGauge common.MemoryGauge) Void {
+	common.UseConstantMemory(memoryGauge, common.MemoryKindCadenceVoid)
+	return NewVoid()
+}
+
 func (Void) isValue() {}
 
-func (Void) Type() Type {
-	return VoidType{}
+func (Void) Type(gauge common.MemoryGauge) Type {
+	return NewVoidType(gauge)
```

Comment:
Maybe we can introduce a new method here as well, keeping the old one as it is, similar to constructors.
Can do it in a subsequent PR.

Reply:
Oh good point - that also alters the interface.

A second PR sounds good to me. This wouldn't be as easy a change as the constructor renamings.

Reply:
https://github.com/dapperlabs/cadence-private-issues/issues/46